File: Day4/catkin_ws/src/afnan/scripts/word_count_client.py
#!/usr/bin/env python

import sys
import rospy
from afnan.srv import wordcount, wordcountRequest, wordcountResponse

def count_word_client(mystring):
    rospy.wait_for_service('count_word')
    try:
        count_word = rospy.ServiceProxy('count_word', wordcount)
        resp1 = count_word(mystring)
        return resp1.count
    except rospy.ServiceException as e:
        print("Service call failed: %s"%e)

def usage():
    return "%s [Input a string]"%sys.argv[0]

if __name__ == "__main__":

    while not rospy.is_shutdown():
        mystring = raw_input ('Enter your string: ')
        print("Requesting number of words in %s"%(mystring))
        print("The number of words is %d"%(count_word_client(mystring)))
        print ("")

File: Day4/catkin_ws/src/afnan/src/wordcount_client.cpp
#include "ros/ros.h"
#include "afnan/wordcount.h"
#include "std_msgs/String.h"
#include "std_msgs/Int64.h"
#include <cstdlib>
#include <string>
#include <iostream>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "count_word_client");

  ros::NodeHandle n;
  ros::ServiceClient client = n.serviceClient<afnan::wordcount>("count_word");
  afnan::wordcount srv;
  std::string MyString = "";
  std::getline (std::cin,MyString);
  srv.request.mystring = MyString;
  std::cout << MyString << "\n";
  std::cout << srv.request.mystring << "\n";
  if (client.call(srv))
  {
    ROS_INFO("Number of words is %d", srv.response.count);
  }
  else
  {
    ROS_ERROR("Failed to call service count_word");
    return 1;
  }

  return 0;
}